images/router/haproxy/conf/haproxy-config.template

```diff
@@ -175,6 +180,10 @@ frontend fe_sni
   bind 127.0.0.1:{{env "ROUTER_SERVICE_SNI_PORT" "10444"}} ssl no-sslv3 {{ if gt (len .DefaultCertificate) 0 }}crt {{.DefaultCertificate}}{{ else }}crt /var/lib/haproxy/conf/default_pub_keys.pem{{ end }} crt-list /var/lib/haproxy/conf/cert_config.map accept-proxy
   mode http
 
+  # DNS labels are case insensitive (RFC 4343), we need to convert the hostname into lowercase
+  # before matching, or any requests containing uppercase characters alawys do not match.
+  http-request set-header Host %[req.hdr(Host),lower]
```

Comment:
as far as I am aware generateRouteRegexp just generates the map file, and the map file matches the values I don't think we can make that match case insensitive. 

you *might* be able to add a case insensitive regex into the map file and change the use_backends to use base_reg instead of base. I think the method in this PR might be cleaner